src/components/icon/icon.component.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import TooltipDecorator from '../../utils/decorators/tooltip-decorator';
+import Icons from './icons';
+import { validProps } from '../../utils/ether';
+import tagComponent from '../../utils/helpers/tags';
+import { StyledIcon, StyledSvgIconWrapper } from './icon.style';
+import OptionsHelper from '../../utils/helpers/options-helper';
+
+const Icon = TooltipDecorator(
+  class Icon extends React.Component {
```

Comment:
Yeah totally agree that this is one of the many examples that are candidates for converting to functional components, especially now we have React Hooks as an option. I think doing it here runs the risk of skewing the scope of the work described in the ticket and would require rewriting a lot of the tests as they target the component instance. I'm going to create a ticket so that it can be carried out as part of some future work

Reply:
I've also just realised after looking into it some more, it would require the `TooltipDecorator` to be rewritten as that requires a class component as a parameter currently

Reply:
OK, sounds like a good plan to create a separate ticket to do this refactoring :thumbsup: